cirq/google/calibration/workflow.py

```diff
@@ -34,16 +223,99 @@ def run_characterizations(
+        results += [
+            calibration.parse_result(result)
+            for calibration, result in zip(calibrations, job.calibration_results())
+        ]
+        if progress_func:
+            progress_func(len(results), len(calibrations))
+
+    return results
+
+
+def run_floquet_characterization_for_circuit(
+    circuit: Circuit,
+    engine: Engine,
+    processor_id: str,
+    gate_set: SerializableGateSet,
+    options: FloquetPhasedFSimCalibrationOptions = FloquetPhasedFSimCalibrationOptions.all_except_for_chi_options(),
+    gates_translator: Callable[[Gate], Optional[FSimGate]] = sqrt_iswap_gates_translator,
+    merge_sub_sets: bool = True,
+    max_layers_per_request: int = 1,
+    progress_func: Optional[Callable[[int, int], None]] = None,
+) -> Tuple[List[PhasedFSimCalibrationResult], List[Optional[int]]]:
+    """Extracts moments within a circuit to characterize and characterizes them against engine.
+
+    The method calls floquet_characterization_for_circuit to extract moments to characterize and
+    run_characterizations to characterize them.
+
+    Args:
+        circuit: Circuit to characterize.
+        engine: cirq.google.Engine object used for running the calibrations.
+        processor_id: processor_id passed to engine.run_calibrations method.
+        gate_set: Gate set to use for characterization request.
+        options: Options that are applied to each characterized gate within a moment. Defaults
+            to all_except_for_chi_options which is the broadest currently supported choice.
+        gates_translator: Function that translates a gate to a supported FSimGate which will undergo
+            characterization. Defaults to sqrt_iswap_gates_translator.
+        merge_sub_sets: Whether to merge moments that can be characterized at the same time
+            together.
+        max_layers_per_request: Maximum number of calibration requests issued to cirq.Engine at a
+            single time. Defaults to 1.
+        progress_func: Optional callback function that might be used to report the calibration
+            progress. The callback is called with two integers, the first one being a number of
+            layers already calibrated and the second one the total number of layers to calibrate.
+
+    Returns:
+        Tuple of:
+          - list of PhasedFSimCalibrationResult for each recognized moment to characterize.
+          - list of indices of the generated characterization requests for each moment in the
+            supplied circuit. If None occurs at certain position, it means that the related moment
+            does not require characterization.
+
+    Raises:
+        IncompatibleMomentError when circuit contains a moment with operations other than the
+        operations matched by gates_translator, or it mixes a single qubit and two qubit gates.
+    """
+    requests, mapping = floquet_characterization_for_circuit(
```

Comment:
it's always a bit weird when `mapping` isn't a dictionary

Reply:
Renamed.